pkg/provider/github/github.go

```diff
@@ -177,3 +186,76 @@ func (v *Provider) getObject(ctx context.Context, sha string, runevent *info.Eve
 	}
 	return decoded, err
 }
+
+// Detect processes event and detect if it is a github event, whether to process or reject it
+// returns (if is a GH event, whether to process or reject, error if any occurred)
+func (v *Provider) Detect(reqHeader *http.Header, payload string, logger *zap.SugaredLogger) (bool, bool, error) {
+	isGH := false
+	event := reqHeader.Get("X-Github-Event")
+	if event == "" {
+		return false, false, nil
+	}
+
+	// it is a Github event
+	isGH = true
+
+	setLoggerAndProceed := func() (bool, bool, error) {
+		logger = logger.With("provider", "github", "event", reqHeader.Get("X-GitHub-Delivery"))
```

Comment:
@chmouel I was passing logger and was hoping to update here 
but it isn't getting updated from where it is called 
I will have to return logger too 🤔 
any better idea ?

Reply:
I am not so sure I understand, what do you mean by updated?